### PR TITLE
Icu 6794 refactor attributes using the newer is nested attribute annotation

### PR DESCRIPTION
**Pull request by ZedLi**
:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-6794)

## Description

:technologist: [Admin preview](https://boundary-ui-git-icu-6794-refactor-attributes-u-0c5223-hashicorp.vercel.app/scopes/s_id6ozx3wue/auth-methods/am_wtbbeizy0q)

This PR aims to remove the usage of attribute fragments in auth methods and hoist them up to the root of the model. This needs to be done before we remove the usage of fragments with the arrays.

There will be a followup PR to cleanup the serializer for normalizing errors.

### CODE REVIEW

File: addons/api/addon/generated/models/auth-method.js [hunk 1]
```diff
@@ -49,4 +49,26 @@ export default class GeneratedAuthMethodModel extends BaseModel {
...
+  @attr('string', { readOnly: true, isNestedAttribute: true }) state;
+
+  @attr('string', { isNestedAttribute: true }) issuer;
+
+  @attr('string', { isNestedAttribute: true }) client_id;
+
+  @attr('string', { isNestedAttribute: true }) client_secret;
+
+  @attr('string', { readOnly: true, isNestedAttribute: true })
+  client_secret_hmac;
```
[the remark this answers is not in the document]
Comment (calcaide) [reply addons/api/addon/generated/models/auth-method.js:62]: Nevermind, that field was part of the fragment, now I see it on the fragment model 😉

**Review by calcaide — APPROVED** [review]
LGTM! Amazing work @ZedLi

**Review by DhariniJeeva — APPROVED** [review]

**Review by randallmorey — APPROVED** [review]
Thanks for the conversion.  I found only attributes tests validating the `state` field.  Should we take this opportunity to add tests for all of the nested attribute fields?

**Comment by ZedLi** [comment]
> Thanks for the conversion. I found only attributes tests validating the `state` field. Should we take this opportunity to add tests for all of the nested attribute fields?

Added more acceptance tests for creating and updating the OIDC form. Also updated the normalizer test to include the fields

**Pull request merged**